src/lib.rs

```diff
@@ -26,18 +22,21 @@ fn run_game_loop(config: &mut Config, guess: char, player_state: &mut PlayerStat
         }
     }
 
-    update_correctly_guessed_letters(config, guess);
+    update_correctly_guessed_letters(config, player_state, guess);
 
-    println!("CORRECT... status: {}", config.correctly_guessed_letters);
+    println!(
+        "CORRECT... status: {}",
+        player_state.correctly_guessed_letters
+    );
 
-    if config.correctly_guessed_letters == config.secret_word {
+    if player_state.correctly_guessed_letters == config.secret_word {
         return GameState::WON;
     }
 
     return GameState::PLAYING;
 }
 
-fn update_correctly_guessed_letters(config: &mut Config, guess: char) {
+fn update_correctly_guessed_letters(config: &Config, player_state: &mut PlayerState, guess: char) {
     // check for hits of guess in solution
```

Comment:
Very nice! I think that this split between static and mutable context is much nicer!